services/121-service/src/metrics/metrics.service.ts

```diff
@@ -1121,17 +1121,58 @@ export class MetricsService {
     };
   }
 
-  private async getCardBalances(programId: number): Promise<{
+  private async createIntersolveVisaBalancesExport(programId: number): Promise<{
     fileName: ExportType;
-    data: any[];
+    data: ExportVisaCardDetails[];
   }> {
-    const data = await this.intersolveVisaExportService.getCards(programId);
+    const rawDebitCardDetails =
+      await this.registrationScopedRepository.getDebitCardsDetailsForExport(
+        programId,
+      );
+
+    const mappedDebitCardDetails =
+      this.mapIntersolveVisaBalancesDataToDto(rawDebitCardDetails);
+
     return {
-      fileName: ExportType.cardBalances,
-      data,
+      fileName: ExportType.intersolveVisaCardDetails,
+      data: mappedDebitCardDetails,
     };
   }
 
+  private mapIntersolveVisaBalancesDataToDto(
+    exportVisaCardRawDetails: ExportVisaCardDetailsRawData[],
+  ): ExportVisaCardDetails[] {
+    let previousRegistrationProgramId: number | null = null;
+    const exportCardDetailsArray: ExportVisaCardDetails[] = [];
+    for (const cardRawData of exportVisaCardRawDetails) {
+      const isCurrentWallet =
+        previousRegistrationProgramId === cardRawData.paId ? false : true;
```

Comment:
I do not understand this code nor the suggestion.

Reply:
Maybe it would help if I were to rewrite it in an even worse but more obvious way:

```ts
let isCurrentWallet;

if (previousRegistrationProgramId === cardRawData.paId) {
  isCurrentWallet = false;
} else {
  isCurrentWallet = true;
}
```

Reply:
Let's have a conversation.

Reply:
Implemented suggestion in new branch.